File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/main/MainView.java
package com.teo.ttasks.ui.activities.main;

import android.support.annotation.NonNull;

import com.teo.ttasks.data.model.TTaskList;
import com.teo.ttasks.ui.base.MvpView;

import java.util.List;

interface MainView extends MvpView {

    /** Called when the user has a profile picture */
    void onUserPicture(@NonNull String pictureUrl);

    /** Called when the user has a cover picture */
    void onUserCover(@NonNull String coverUrl);

    void onTaskListsLoaded(List<TTaskList> taskLists, int currentTaskListIndex);

    void onTaskListsLoadError();
}

File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/main/MainView.kt
package com.teo.ttasks.ui.activities.main

import com.teo.ttasks.data.model.TaskList
import com.teo.ttasks.ui.base.MvpView

interface MainView : MvpView {

    /** Called when the user has a cover picture  */
    fun onUserCover(coverUrl: String)

    fun onTaskListsLoaded(taskLists: List<TaskList>, currentTaskListIndex: Int)

    fun onTaskListsLoadError()

    fun onSignedOut()
}
